File: #858/number15.php
<?php

$countNumbers = intval(trim(fgets(STDIN)));
$finalNumber = 0;
for ($i = 0; $i < $countNumbers; $i++) {
	$number = intval(trim(fgets(STDIN)));
	if (!($number % 2)) {
		if ($number < $finalNumber || $finalNumber === 0) {
			$finalNumber = $number;
		}
	}
}
echo($finalNumber);
?>

File: #858/number15.cpp
#include <iostream>

using namespace std;

int main() {
	int countNumbers, finalNumber, number;

	cin >> countNumbers;

	for (int i = 0; i < countNumbers; i++) {
		cin >> number;
		if (!(number % 2)) {
			if (finalNumber == 0 || number < finalNumber) {
				finalNumber = number;
			}
		}
	}
	cout << finalNumber << endl;
	return 0;
}